datalad/cmdline/main.py

```diff
@@ -205,9 +205,8 @@ def setup_parser(
                   parsed_args, unparsed_args)
     except Exception as exc:
         lgr.debug("Early parsing failed with %s", exc_str(exc))
-        fail_handler(parser)
         need_single_subparser = False
-
+        unparsed_args = cmdlineargs[1:]  # referenced before assignment otherwise
```

Comment:
it is pity there is no unittest which demonstrates the problem and the fix... I am afraid I would break it again at some point in the future.

Academically, it is not clear to me why `parser` went all the way to that `-l` since it didn't even know about `run`, so I expected it to fail at that point already, leaving `-l` etc in unparsed_args.

Reply:
Well, academically it is not clear to me, why you use a private method of a class from a foreign package, while not being clear about what exactly it does. :-P ;-)